pkg/kv/kvserver/replica_application_cmd.go

```diff
@@ -44,8 +42,7 @@ import (
 // batch's view of ReplicaState. Then the batch is committed, the side-effects
 // are applied and the local result is processed.
 type replicatedCmd struct {
-	ent              *raftpb.Entry // the raft.Entry being applied
-	decodedRaftEntry               // decoded from ent
+	*raftlog.Entry
```

Comment:
There is a `raftentry` package. Not sure it's about the same thing, but maybe there should be some integration with it in a follow-up to this PR.

Reply:
I have a hard time waterfalling the eventual package structure, so for now I'm happy just making any refactor that squarely moves us towards a better organization of the codebase. `raftentry` currently houses the `raftentry.Cache` and it makes sense that it will ultimately somehow live closer to the raft log.
I'd like to spend some time pulling more components out of `kvserver` to unblock the pending work though, before focusing on how all of these leaf packages should be organized.

Reply:
Left a TODO comment as discussed in DM.